### PR TITLE
Show error on validate TOTP form with empty answer

### PR DESCRIPTION
**Pull request by alexeystadnikov-okta**
screen shot 2017-08-22 at 6 37 47 pm

screen shot 2017-08-22 at 6 43 43 pm

@mauriciocastillosilva-okta 
@yiyangli-okta 

Please, review.

### CODE REVIEW

File: src/models/Factor.js [hunk 1]
```diff
@@ -168,10 +168,10 @@ function (Okta, Q, factorUtil, BaseLoginModel) {
           return (provider === 'OKTA' && _.contains(['push', 'sms', 'call', 'email'], factorType));
         }
       },
-      isSMSCallorEmail: {
+      needNotEmptyAnswer: {
```
Comment (mauriciocastillosilva-okta) [inline src/models/Factor.js]: we can probably rename this to ```isAnswerRequired```

Comment (alexeystadnikov-okta) [reply src/models/Factor.js]: Done

**Comment by alexeystadnikov-okta** [comment]
Tests for SMS and Call factors added.

**Comment by alexeystadnikov-okta** [comment]
@mauriciocastillosilva-okta
@yiyangli-okta

Any other comments?

File: test/unit/spec/MfaVerify_spec.js [hunk 2]
```diff
@@ -690,6 +690,22 @@ function (Okta,
...
           });
         });
+        itp('shows errors if verify button is clicked and answer is empty', function () {
+          return setupSecurityQuestion()
+          .then(function (test) {
+            $.ajax.calls.reset();
+            test.setNextResponse(resSuccess);
+            test.form.setAnswer('');
+            test.form.submit();
+            return tick(test);
```
Comment (mauriciocastillosilva-okta) [inline test/unit/spec/MfaVerify_spec.js]: we should change this to ```return Expect.waitForFormError(test.form, test);```

File: test/unit/spec/MfaVerify_spec.js [hunk 3]
```diff
@@ -690,6 +690,22 @@ function (Okta,
             expect(test.form.errorMessage()).toBe('Your answer doesn\'t match our records. Please try again.');
           });
         });
+        itp('shows errors if verify button is clicked and answer is empty', function () {
+          return setupSecurityQuestion()
+          .then(function (test) {
+            $.ajax.calls.reset();
+            test.setNextResponse(resSuccess);
```
Comment (mauriciocastillosilva-okta) [inline test/unit/spec/MfaVerify_spec.js]: we probably don't need to set a next response, since we won't make a request.

Comment (alexeystadnikov-okta) [reply test/unit/spec/MfaVerify_spec.js]: Done

File: test/unit/spec/MfaVerify_spec.js (hunk 2 above)
Comment (alexeystadnikov-okta) [reply test/unit/spec/MfaVerify_spec.js]: Done

**Pull request merged**